software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakClusterImpl.java

```diff
@@ -91,8 +94,16 @@ public void start(Collection<? extends Location> locations) {
     }
 
     protected EntitySpec<?> getMemberSpec() {
-        return getConfig(MEMBER_SPEC, EntitySpec.create(RiakNode.class));
-
+        if(getConfig(SEARCH)) {
+            return getConfig(
+                    MEMBER_SPEC,
+                    EntitySpec.create(SameServerEntity.class)
```

Comment:
`SameServerEntity` breaks the assumptions made by the entity. See previous comments.